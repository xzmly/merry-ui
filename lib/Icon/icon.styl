.merry-ui-icon  width: 10em; height: 10em;  vertical-align: -0.15em;  fill: currentColor;  overflow: hidden;