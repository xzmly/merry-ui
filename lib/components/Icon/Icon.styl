@import '../../helpers/stylus.helper.styl';.{class-prefix}-icon  font-size min-font-size  width: 1em; height: 1em;  vertical-align: -0.15em;  fill: currentColor;  overflow: hidden;