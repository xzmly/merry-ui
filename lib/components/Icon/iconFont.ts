interface IconFont {    config: string;}const customCache = new Set<string>();const creatIconFont = (arg: IconFont): void => {    const {config} = arg;    if (        !customCache.has(config) &&        config.length    ) {        const script: HTMLElement = document.createElement('script');        script.setAttribute('src', config);        script.setAttribute('data-namespace', config);        customCache.add(config);        document.body.appendChild(script);    }};export default creatIconFont;