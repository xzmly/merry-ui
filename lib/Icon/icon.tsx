import * as React from 'react'
import classes from "../helpers/classes"
import "./icon.styl"
import '../helpers/importIcons';

interface IconProps extends React.SVGAttributes<SVGElement> {
  name: string;
}

const Icon: React.FunctionComponent<IconProps> =
    ({className,name,...restProps}) => {
  return (
      <svg className={`${classes(className,'icon')}`}
           {...restProps} >
        <use xlinkHref={`#${name}`}/>
      </svg>
  )
}

export default Icon
