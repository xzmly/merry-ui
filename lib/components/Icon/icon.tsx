import * as React from 'react';
import classes from '../../helpers/classes';
import './icon.styl';
import '../../helpers/importIcons';
import createIconFont from './iconFont';

interface IconProps extends React.SVGAttributes<SVGElement> {
    name: string;
}

interface IconComponent<P> extends React.FC<P> {
    createIconFont: typeof createIconFont
}

const Icon: IconComponent<IconProps> =
    ({className, name, ...restProps}) => {
        return (
            <svg className={`${classes(className, 'icon')}`}
                 {...restProps} >
                <use xlinkHref={`#${name}`}/>
            </svg>
        );
    };

Icon.createIconFont = createIconFont;

export default Icon;
