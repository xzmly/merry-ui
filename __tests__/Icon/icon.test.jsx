import renderer from 'react-test-renderer'
import Icon from '../../lib/Icon/icon'
import React from 'react'

describe('icon',()=>{
  it('不是个div',()=>{
    const json = renderer.create(<Icon/>).toJSON()
    expect(json).toMatchSnapshot()
  })
})