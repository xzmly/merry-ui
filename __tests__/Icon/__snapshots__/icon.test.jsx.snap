// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon 不是个div 1`] = `
<div
  className="icon"
/>
`;
